common/src/main/java/tachyon/LeaderSelectorClient.java

```diff
@@ -143,10 +143,17 @@ public void stateChanged(CuratorFramework client, ConnectionState newState) {
 
   @Override
   public void takeLeadership(CuratorFramework client) throws Exception {
+    List<String> childList = new ArrayList<String>();
     mIsLeader.set(true);
-    if (client.checkExists().forPath(mLeaderFolder + mName) != null) {
-      client.delete().forPath(mLeaderFolder + mName);
+    if (mLeaderFolder != null && client.checkExists()
```

Comment:
Could you add comments for these changes. Also feel free to extract it into a function, if that could better describe the behavior.